packages/shared/src/lib.ts

```diff
@@ -1,6 +1,22 @@
-export const USER_EVENTS = {
+export const UserEvents = {
   UsernameCreated: 'username-created',
   UsernameMissing: 'username-missing',
 } as const;
+export type UserEvents = typeof UserEvents[keyof typeof UserEvents];
 
-export type UserEvents = typeof USER_EVENTS[keyof typeof USER_EVENTS];
+export const GameState = {
+  Joined: 'joined',
```

Comment:
If this a game state, then it's probably should be something like `Started` or `Active` or something like that, right? From what I see, you differentiated between Game (where game login/rendering is going to be) and Room (where are things relative to users joining a room with a game).

Reply:
happy to refactor but right now I don't have a clear picture on what it should be. This is not probably game state but something that connect player to room? Not sure in that sense is connected to a game but since we didn't think how the model will be 🤷‍♂️

Reply:
Sure, we can have it room state then...? Whatever, it's fine for now, yes.